app/src/main/java/io/rapidpro/androidchannel/payload/MTBroadcast.java

```diff
@@ -77,6 +77,12 @@ public void execute(Context context, SyncPayload sent){
                     RapidPro.LOG.d("Resending DELIVERED notification for " + msg.getPhone() + " id: " + to.getId());
                     DBCommandHelper.queueCommand(context, new MTTextDelivered(to.getId(), msg.getPhone()));
                 }
+
+                // Resend from server
+                if (state == MTTextMessage.FAILED || state == MTTextMessage.FAILED_SYNCED) {
+                    RapidPro.LOG.d("Server request to resend for " + msg.getPhone() + " id: " + to.getId());
+                    DBCommandHelper.updateCommandStateWithServerId(context, MTTextMessage.CMD, to.getId(), MTTextMessage.RETRY, "" + 0);
+                }
```

Comment:
We use to rely on the fact the resent messages used new ids, now that we use the same IDs, we need to update and set the retries back to 0 on the same command